send-transaction-service/src/send_transaction_service.rs

```diff
@@ -305,12 +464,28 @@ impl SendTransactionService {
             }
         });
 
+        if !batched_transactions.is_empty() {
+            // Processing the transactions in batch
+            let addresses = Self::get_tpu_addressesn(tpu_address, leader_info, config);
+
+            let wire_transactions = transactions
+                .iter()
+                .filter(|(signature, _)| batched_transactions.contains(signature))
+                .map(|(_, transaction_info)| transaction_info.wire_transaction.as_ref())
+                .collect::<Vec<&[u8]>>();
+
+            for address in &addresses {
+                let iter = wire_transactions.chunks(config.batch_size);
+                for chunk in iter {
+                    Self::send_transactions(address, chunk);
+                }
+            }
```

Comment:
are we sure this will send fast enough to safely reuse the leader info in all cases?

Reply:
The leader info is refreshed in the retry_thread periodically. My limited testing with UDP seems to be holding pretty well even with 100 threads concurrently pounding on it. We know quic has its own issue even without batching. That said, the change in here did not introduce side effect of the validity of leader info logic. It may influence the size in the retry set -- the test so far seems to indicate it helps to reduce it overall size faster. The worst case scenario net impact is the batch size of transactions being all retried which I think is okay after testing. The previous implementation also refresh the leader info periodically and use that to retry the transactions for all of those eligible for retry. In a busy system the size in the retry transaction set can also be well beyond the batch size in the previous implementation.

Reply:
Understood the concern. I think the issue can be tackled by a leader info refresher object. I will address this issue in follow-on PR after this as this PR is growing larger.